march_rqt_robot_monitor/src/march_rqt_robot_monitor/diagnostic_analyzers/temperature.py

```diff
@@ -0,0 +1,85 @@
+
+from diagnostic_msgs.msg import DiagnosticStatus
+import rospy
+from sensor_msgs.msg import Temperature
+
+
+LOWER_THRESHOLD_VALID_VALUE = 0
+UPPER_THRESHOLD_VALID_VALUE = 100
+
+
+class CheckJointTemperature(object):
+    """Base class to diagnose the joint temperatures."""
+
+    def __init__(self, joint_name, topic):
+        rospy.Subscriber(topic, Temperature, self.cb)
+
+        self._default_thresholds = \
+            rospy.get_param('/safety_node/default_temperature_threshold', None)
+        self._thresholds_warning = \
+            rospy.get_param('/safety_node/temperature_thresholds_warning/{jn}'.format(jn=joint_name), None)
+        self._thresholds_non_fatal = \
+            rospy.get_param('/safety_node/temperature_thresholds_non_fatal/{jn}'.format(jn=joint_name), None)
+        self._thresholds_fatal = \
+            rospy.get_param('/safety_node/temperature_thresholds_fatal/{jn}'.format(jn=joint_name), None)
```

Comment:
I think for this path to always work you have to remove the preceding '/'. That means it will always look in the root of the parameter server. However, the march nodes, when launched together, will be placed in the `march` namespace. So in order to always look relative to this node's current namespace you must remove the `/`.
```suggestion
        self._default_thresholds = \
            rospy.get_param('/safety_node/default_temperature_threshold', None)
        self._thresholds_warning = \
            rospy.get_param('safety_node/temperature_thresholds_warning/{jn}'.format(jn=joint_name), None)
        self._thresholds_non_fatal = \
            rospy.get_param('safety_node/temperature_thresholds_non_fatal/{jn}'.format(jn=joint_name), None)
        self._thresholds_fatal = \
            rospy.get_param('safety_node/temperature_thresholds_fatal/{jn}'.format(jn=joint_name), None)
```